### ISSUE TITLE
enable config option for set_bold_is_bright

### ISSUE THREAD

**Issue by M0M097** [cdepillabout/termonad#176]
In almost every terminal emulator bold text uses bright colors. A option to enable this or some help to do that would be great. 

The default bold blue text is almost impossible to read.

**Comment by cdepillabout** [comment]
@M0M097 Could you post a screen shot that doesn't look good to you?

Font weight (bold) and colors are orthogonal as far as most terminals go.  You can set bold vs. not bold completely separate from whether you use the basic or extended palette:

https://hackage.haskell.org/package/termonad-4.0.1.2/docs/Termonad-Config-Colour.html#t:Palette

------------------------------------------------------

> The default bold blue text is almost impossible to read.

I agree that this is a problem, but I think this is a problem with blue text in general, not specifically bold blue text.  Also, this is a problem on most terminals (well, at least VTE terminals because of the defaults).

In my personal Termonad config, I change the color of both the basic and extended blue to make it lighter:

```console
{-# LANGUAGE TypeApplications #-}
{-# LANGUAGE OverloadedStrings #-}

module Main where

import Control.Lens ((&), set)
import Data.Colour.SRGB (sRGB24)
import Termonad.App (defaultMain)
import Termonad.Config
import Termonad.Config.Colour

main :: IO ()
main = do
  let colConf =
        defaultColourConfig
          { cursorBgColour = Set $ createColour 204 0 0
          , palette =
              let myStandardColors =
                    setAtList8 4 (createColour 120 120 250) $ -- this is the important line
                    defaultStandardColours
                  myLightCols =
                    setAtList8 4 (createColour 150 150 250) $ -- this is the important line
                    defaultLightColours
              in ExtendedPalette myStandardColors myLightCols
          }
  colExt <- createColourExtension colConf
  let tmConf =
        defaultTMConfig
          { options =
              defaultConfigOptions
                { fontConfig =
                    FontConfig
                      { fontFamily = "DejaVu Sans Mono"
                      , fontSize = FontSizePoints 13
                      }
                , showScrollbar = ShowScrollbarAlways
                , showTabBar = ShowTabBarAlways
                , scrollbackLen = 20000
                }
          }
        `addColourExtension` colExt
  defaultMain tmConf
```

It looks like this:

![image](https://user-images.githubusercontent.com/64804/103470572-1fa20d80-4db7-11eb-8a80-1c44cd4f57ea.png)

(This screen shot was generated with https://hackage.haskell.org/package/print-console-colors.)

Here's what the default VTE colors look like:

![image](https://user-images.githubusercontent.com/64804/103470591-760f4c00-4db7-11eb-9200-d3d5de93486a.png)

In my opinion, basic (standard) blue is too dark here.

**Comment by M0M097** [comment]
Keep in mind that just changing the main blue color may result in ugly tui applications. 

I'm aware that boldness and bright color are separated things but usually they are combined in terminals since back in the day \033[1; was setting bold and bright together. In the mentioned vte this is achieved by  `set_bold_is_bright(true)`. And in st there is the following code:

`/* Change basic system colors [0-7] to bright system colors [8-15] */
	if ((base.mode & ATTR_BOLD_FAINT) == ATTR_BOLD && BETWEEN(base.fg, 0, 7))
		fg = &dc.col[base.fg + 8];`

which does the same. If this behavior is not wanted as default that's fine, but there should be a way to enable this in the termonad.hs file.

**Comment by cdepillabout** [comment]
@M0M097 Ah, thanks for the pointer to `set_bold_is_bright`.

It looks like `gi-vte` exposes this option as well: http://hackage.haskell.org/package/gi-vte-2.91.27/docs/GI-Vte-Objects-Terminal.html#v:terminalSetBoldIsBright

If you (or anyone else) wanted to send a PR adding this option, I'd be happy to merge it in.

I imagine you'd need to add it to the Termonad config in this file:

https://github.com/cdepillabout/termonad/blob/d8998e6f30b86103b55fa3117cf0f6f4c24b7e07/src/Termonad/Types.hs#L368

And then based on this option, you could call `terminalSetBoldIsBright` here:

https://github.com/cdepillabout/termonad/blob/d8998e6f30b86103b55fa3117cf0f6f4c24b7e07/src/Termonad/Term.hs#L342

-------------------------------

I'll create a separate issue for potentially changing the default blue color.

**Comment by M0M097** [comment]
Hardcoding `terminalSetBoldIsBright vtTerm True` in termonad/src/Termonad/Term.hs works fine so it should be easy to add an option to Types.hs .

I tried to implement that in the same way as `confirmExit` is implemented but I'm getting a parsing error if I try to compile it. I basically don't know any Haskell so I must have made an obvious mistake.

Anyway, thanks for your suggestions. With hardcoding this option my termonad now displays readable bold text and the standard colors are still available for tui applications, which was everything I was looking for :)

**Comment by cdepillabout** [comment]
Implemented in #178
